File: problems/Java/71. Simplify Path.java
class Solution {
    public String simplifyPath(String path) {
        Deque<String> stk = new ArrayDeque<>();
        String[] comps = path.split("/");

        for (String comp : comps) {
            if (comp.equals("..")) {
                if (!stk.isEmpty()) {
                    stk.pop();
                }
            } else if (!comp.isEmpty() && !comp.equals(".")) {
                stk.push(comp);
            }
        }

        if (stk.isEmpty()) {
            return "/";
        }

        StringBuilder res = new StringBuilder();
        while(!stk.isEmpty()){
            res.insert(0, stk.pop()).insert(0, "/");
        }
        return res.toString();
    }
}

File: problems/Python/71. Simplify Path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stk = []
        for comp in path.split('/'):
            if comp == '..':
                if stk:
                    stk.pop()
            elif comp and comp != '.':
                stk.append(comp)
        
        return '/' + '/'.join(stk)